lib/usagereporter/teleport/usagereporter.go

```diff
@@ -122,12 +122,16 @@ func (t *StreamingUsageReporter) Run(ctx context.Context) {
 
 type SubmitFunc = usagereporter.SubmitFunc[prehogv1a.SubmitEventRequest]
 
-func NewStreamingUsageReporter(log logrus.FieldLogger, clusterName types.ClusterName, submitter SubmitFunc) (*StreamingUsageReporter, error) {
+func NewStreamingUsageReporter(log logrus.FieldLogger, clusterName types.ClusterName, anonymizationKey string, submitter SubmitFunc) (*StreamingUsageReporter, error) {
 	if log == nil {
 		log = logrus.StandardLogger()
 	}
+	anonKey := anonymizationKey
```

Comment:
WDYT about removing the `clusterName` argument and letting the caller determine whether to use the cluster ID or something else for anonymization?

As written, we require the caller to provide both fields even though only one will ever be used.

Reply:
`clusterName` argument is used later here, so can't be removed: https://github.com/gravitational/teleport/blob/9c4fa206c9868c61052d6a13c50a64c63cdea602/lib/usagereporter/teleport/usagereporter.go#L110-L117

Moved fallback logic from anonymizationKey to clusterId to caller side (in teleport.e side)